docs/language-server/protocol-language-server.md

```diff
@@ -2912,6 +2913,46 @@ root_type VisualisationUpdate;
 
 N/A
 
+### `executionContext/visualisationEvaluationFailed`
+
+Signals that an evaluation of a visualisation expression on the computed value
+has failed.
+
+- **Type:** Notification
+- **Direction:** Server -> Client
+- **Connection:** Protocol
+- **Visibility:** Public
+
+#### Parameters
+
+```typescript
+interface VisualisationEvaluationFailed {
+  /**
+   * An execution context identifier.
+   */
+  contextId: ContextId;
+
+  /**
+   * A visualisation identifier.
+   */
+  visualisationId: UUID;
+
+  /**
+   * An identifier of a visualised expression.
+   */
+  expressionId: UUID;
+
+  /**
+   * An error message.
+   */
+  message: string;
```

Comment:
Surely we want to provide richer error information than just a string? We have a whole API for providing detailed error information with locations, traces, and so on. Given that errors are first class in Enso, and visualisation preprocessors are Enso code, their errors should be reported with the same detail as an error in the main language.

Reply:
I added diagnostic info. The same error details that we send in case of execution errors.